egs/librispeech/s5/local/rnnlm/tuning/run_tdnn_lstm_1a.sh

```diff
@@ -0,0 +1,144 @@
+fi
+
+if [ $stage -le 1 ]; then
+  cp $lexicon $dir/config/
+  n=`cat $dir/config/words.txt | wc -l`
+  echo "<brk> $n" >> $dir/config/words.txt
+
+  # words that are not present in words.txt but are in the training or dev data, will be
+  # mapped to <SPOKEN_NOISE> during training.
+  echo "<UNK>" >$dir/config/oov.txt
+
+  cat > $dir/config/data_weights.txt <<EOF
+librispeech   1   1.0
+EOF
+
+  rnnlm/get_unigram_probs.py --vocab-file=$dir/config/words.txt \
+                             --unk-word="<UNK>" \
+                             --data-weights-file=$dir/config/data_weights.txt \
+                             $text_dir | awk 'NF==2' >$dir/config/unigram_probs.txt
+
+  # choose features
+  rnnlm/choose_features.py --unigram-probs=$dir/config/unigram_probs.txt \
+                           --top-word-features=5000 \
+                           --use-constant-feature=true \
+                           --special-words='<s>,</s>,<brk>,<UNK>,<SPOKEN_NOISE>' \
+                           $dir/config/words.txt > $dir/config/features.txt
+
+  cat >$dir/config/xconfig <<EOF
+input dim=$embedding_dim name=input
+relu-renorm-layer name=tdnn1 dim=$embedding_dim input=Append(0, IfDefined(-1))
+fast-lstmp-layer name=lstm1 cell-dim=$embedding_dim recurrent-projection-dim=$lstm_rpd non-recurrent-projection-dim=$lstm_nrpd
+relu-renorm-layer name=tdnn2 dim=$embedding_dim input=Append(0, IfDefined(-3))
+fast-lstmp-layer name=lstm2 cell-dim=$embedding_dim recurrent-projection-dim=$lstm_rpd non-recurrent-projection-dim=$lstm_nrpd
+relu-renorm-layer name=tdnn3 dim=$embedding_dim input=Append(0, IfDefined(-3))
+output-layer name=output include-log-softmax=false dim=$embedding_dim
+EOF
+  rnnlm/validate_config_dir.sh $text_dir $dir/config
+fi
+
+if [ $stage -le 2 ]; then
+  # the --unigram-factor option is set larger than the default (100)
+  # in order to reduce the size of the sampling LM, because rnnlm-get-egs
+  # was taking up too much CPU (as much as 10 cores).
+  rnnlm/prepare_rnnlm_dir.sh --unigram-factor 400 \
+                            $text_dir $dir/config $dir
+fi
+
+if [ $stage -le 3 ]; then
+  rnnlm/train_rnnlm.sh --num-jobs-final 2 \
+                       --stage $train_stage \
+                       --num-epochs $epochs \
+                       --cmd "$train_cmd" $dir
+fi
+
+LM=fglarge 
+if [ $stage -le 4 ] && $run_lat_rescore; then
+  echo "$0: Perform lattice-rescoring on $ac_model_dir"
+#  LM=tgsmall # if using the original 3-gram G.fst as old lm
+  pruned=
+  if $pruned_rescore; then
+    pruned=_pruned
+  fi
+  for decode_set in test_clean test_other dev_clean dev_other; do
+    decode_dir=${ac_model_dir}/decode_${decode_set}_${LM}
+
+    # Lattice rescoring
+    rnnlm/lmrescore$pruned.sh \
+      --cmd "$decode_cmd --mem 8G" \
+      --weight 0.4 --max-ngram-order $ngram_order \
+      data/lang_test_$LM $dir \
+      data/${decode_set}_hires ${decode_dir} \
+      exp/decode_${decode_set}_${LM}_${decode_dir_suffix}_rescore
```

Comment:
@keli78 put the rescore dir under exp? or for example under exp/chain_cleaned/tdnn_lstm_1n_sp/?

Reply:
will modify this when I update the WERs. Thanks!